.github/workflows/ci.yml

```diff
@@ -27,13 +26,50 @@ jobs:
           node-version: 20
       - run: npm ci
       - run: npm test
-  deploy:
+  build-publish:
+    runs-on: ubuntu-latest
+    services:
+      postgres:
+        image: postgres:latest
+        env:
+          POSTGRES_DB: postgres
+          POSTGRES_PASSWORD: postgres
+          POSTGRES_USER: postgres
+        ports:
+          - 5432:5432
+        options: >-
+          --health-cmd pg_isready
+          --health-interval 10s
+          --health-timeout 5s
+          --health-retries 5
+    env:
+      DATABASE_URL: postgres://postgres:postgres@localhost:5432/postgres
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-node@v3
+        with:
+          node-version: 20
+      - run: npm ci
+      - run: npm run migrate
+      - run: cd spark-publish && npm ci
+      - run: cd spark-publish && npm test
+  deploy-api:
     if: github.ref == 'refs/heads/main'
-    needs: build
+    needs: [build-api, build-publish]
```

Comment:
IIUC, this config allows deployments of new spark-api versions only if both spark-api and spark-publish pass the tests. (And vice versa for deploying new spark-publish versions.)

The other option is to allow spark-api deployments as long as spark-api tests are passing, and similarly for spark-publish deployments.

I am fine either way; just wanted to make sure this is a conscious decision on our side.

Reply:
The primary reason for having both projects in one repo is that they will always be deployed together, that's why I set it up like this. If we are fine with deploying them separately, I'd move them to different repos. Wdyt?